src/mscorlib/src/System/MulticastDelegate.cs

```diff
@@ -103,11 +104,22 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
         [System.Security.SecuritySafeCritical]  // auto-generated
         public override sealed bool Equals(Object obj)
         {
-            if (obj == null || !InternalEqualTypes(this, obj))
-                return false;      
-            MulticastDelegate d = obj as MulticastDelegate;
-            if (d == null)
+            if (obj == null)
                 return false;
+            if (object.ReferenceEquals(this, obj))
+                return true;
+            
+            // TODO: See if this can be changed to GetType() == obj.GetType()
+            // without negatively impacting performance. GetType is a JIT
```

Comment:
@jkotas I removed the TODO. (It seems kinda unintuitive that you have to define a new function/ P/Invoke to get faster performance than what's written 99% of the time, but whatever.)